### ISSUE TITLE
Length cannot be less than zero when converting policy to rules

### ISSUE THREAD

**Issue by BernieWhite** [Azure/PSRule.Rules.Azure#1802]
**Description of the issue**

When calling `Export-AzPolicyAssignmentRuleData` an exception is raised.

**To Reproduce**

Steps to reproduce the issue:

```powershell
Export-AzPolicyAssignmentRuleData -AssignmentFile .\tests\PSRule.Rules.Azure.Tests\Policy.assignment.json
```

**Error output**

```text
Export-AzPolicyAssignmentRuleData: Length cannot be less than zero. (Parameter 'length')
```

**Module in use and version:**

- Module: PSRule.Rules.Azure
- Version: **1.21.0-B0050**

Captured output from `$PSVersionTable`:

```text
Name                           Value
----                           -----
PSVersion                      7.2.7
PSEdition                      Core
GitCommitId                    7.2.7
OS                             Microsoft Windows 10.0.22621
Platform                       Win32NT
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0…}
PSRemotingProtocolVersion      2.3
SerializationVersion           1.1.0.1
WSManStackVersion              3.0
```
